Setup/Patch/Data/MigrateVirtualReplicaConfigPatch.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Setup\Patch\Data;
+
+use Algolia\AlgoliaSearch\Api\Product\ReplicaManagerInterface;
+use Algolia\AlgoliaSearch\Exception\ReplicaLimitExceededException;
+use Algolia\AlgoliaSearch\Exception\TooManyCustomerGroupsAsReplicasException;
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Algolia\AlgoliaSearch\Helper\Configuration\ConfigChecker;
+use Algolia\AlgoliaSearch\Service\Product\SortingTransformer;
+use Algolia\AlgoliaSearch\Service\StoreNameFetcher;
+use Algolia\AlgoliaSearch\Validator\VirtualReplicaValidatorFactory;
+use Magento\Framework\App\Config\ConfigResource\ConfigInterface;
+use Magento\Framework\App\Config\ReinitableConfigInterface;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\App\Config\Storage\WriterInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Serialize\SerializerInterface;
+use Magento\Framework\Setup\ModuleDataSetupInterface;
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+use Magento\Framework\Setup\Patch\PatchInterface;
+use Magento\Store\Model\ScopeInterface;
+use Magento\Store\Model\StoreManagerInterface;
+
+class MigrateVirtualReplicaConfigPatch implements DataPatchInterface
```

Comment:
Is the order of patch application important somehow between the 2? How is it defined ?

Reply:
Yes! This patch should run first. Sequence is handled via the dependency chain. The second patch [will not run](https://github.com/algolia/algoliasearch-magento-2/pull/1562/files#diff-26b7610beb757502880d5480b94e43e1f54371981936cc24736c083400f4dab8R29) until this one has.